tests/unit/auth/test_signers.py

```diff
@@ -290,6 +290,31 @@ def test_query_string_params_in_urls(self):
         self.assertEqual('marker=%C3%A4%C3%B6%C3%BC-01.txt&prefix=', cqs)
 
 
+class TestSigV4(unittest.TestCase):
+    def setUp(self):
+        self.credentials = botocore.credentials.Credentials(
+            access_key='foo', secret_key='bar')
+
+    def test_canonical_query_string(self):
+        request = AWSRequest()
+        request.url = (
+            'https://search-testdomain1-j67dwxlet67gf7ghwfmik2c67i.us-west-2.'
+            'cloudsearch.amazonaws.com/'
+            '2013-01-01/search?format=sdk&pretty=true&q=George%20Lucas&'
```

Comment:
Is this query params already sorted here? It may be beneficial to rearrange some of them to ensure they get sorted.